### PR TITLE
Fix pytorch related pytest failures

### PR DESCRIPTION
**Pull request by JanFSchulte**
There currently are a couple of pytorch-related failures in the pytests. 

The main one is a bug with recurrent layers where the weights and hidden weights need to be transposed to mach the keras convention that is used in our implementation. Since this is not strictly a channels-last conversion I have added this directly to the parser so that the channels-last conversion feature can still be used as before. This issue was masked before because of a bug in the converter that ignored the "off" setting. After I fixed this recently, this issue was revealed, but missed at the time. 

I also increased the bit width in one of the einsum tests to avoid occasional failures because of limited precision. There is nothing wrong otherwise, so this seems like the best option. 

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)

## Tests

Failing pytests pass now

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/main/CONTRIBUTING.md).
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have installed and run `pre-commit` on the files I edited or added.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by calad0i** [comment]
Can you push an empty commit and rerun the tests?

**Comment by jmitrevs** [comment]
I retried the HGQ + DA tests. Try to replicate locally with the same random seed. Otherwise, could there be version differences between the test environment and some libraries that would make the difference?

**Comment by JanFSchulte** [comment]
It's certainly reproduced in the test environment here, see for example the oneAPI accelerator PR tests: https://gitlab.cern.ch/fastmachinelearning/hls4ml/-/jobs/60904849 So I guess it's just a matter of luck reproducing it online. I just didn't want to run all 386 tests offline to reproduce on failure and haven't found a seed yet that gives me a failure for a single test.

**Comment by JanFSchulte** [comment]
Guess it passed now. A little weird.

**Review by calad0i — APPROVED** [review]
